### PR TITLE
[2.x] Updated actions config

### PR DESCRIPTION
**Pull request by GrahamCampbell**
To be the same as the framework repos.

### CODE REVIEW

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -25,21 +25,18 @@ jobs:
...
           php-version: ${{ matrix.php }}
           extensions: dom, curl, libxml, mbstring, zip
+          tools: composer:v2
           coverage: none
 
       - name: Install dependencies
-        run: composer require "illuminate/contracts=${{ matrix.laravel }}" --prefer-dist --no-interaction
+        run: |
+          composer require "illuminate/contracts=${{ matrix.laravel }}" --no-update
+          composer update --prefer-dist --no-interaction --no-progress
```
Comment (driesvints) [inline .github/workflows/tests.yml:39]: This isn't necessary for this repo. Can you revert this?

**Comment by driesvints** [comment]
I think it's best that we wait until a stable composer 2 version lands and becomes the new default. Otherwise we'll be update all repos again soon to remove the "2" indicator. Tests are running fast enough for now.

File: .github/workflows/tests.yml (hunk 1 above)
Comment (GrahamCampbell) [reply .github/workflows/tests.yml:39]: Yes, this is necessary, and I can't revert. See the first commit I tried where the tests failed.

**Comment by GrahamCampbell** [comment]
> I think it's best that we wait until a stable composer 2 version lands and becomes the new default. Otherwise we'll be update all repos again soon to remove the "2" indicator. Tests are running fast enough for now.

Actually, the tests will break when composer 2 becomes the default (because of the changes needed to the install step), so this PR is a proactive fix.

**Comment by driesvints** [comment]
@GrahamCampbell so composer 2 will break all apps/servers that auto update composer?

**Comment by GrahamCampbell** [comment]
No. Most apps don't do a require without there already being a lock file present. That is what is not supported. You must ask for either no-update, or already have a lock file.

**Comment by GrahamCampbell** [comment]
What I am talking about is this:

image

**Comment by driesvints** [comment]
Gotcha. I'll try to look into this some more next week.

**Comment by GrahamCampbell** [comment]
Perfect. :)

**Pull request merged**